### PR TITLE
Create pr-check.yml

### LINKED ISSUE #41: Add GitHub Actions workflow for testing on every PR
Tests should be ran on every Pull Request. This will allow for:
- Blocking merges if tests are failing
- Will send clear message to contributors and maintainers that there's an issue (and often will hint where it's at)

Tests should be ran only for Pull Requests and not for merges to `main` branch as PRs are the only way to commit to `main`.

This could be a starting point for `.github/workflows/pr-check.yml` file:
```yml
name: Test PR

on:
  pull_request:
    branches:
      - main

jobs:
  check-pr:
    runs-on: ubuntu-latest

    steps:
      - name: Checkout code
        uses: actions/checkout@v3

      - name: Set up Go
        uses: ???
      
      - name: Run tests
        uses: ???
```

### PR DESCRIPTION
**Pull request by Pr0-C0der**
Created Github actions workflow for Fixes #41. 

While testing runs all files with the suffix "_test.go", and does not merge the PR if any test fails. 

To create a message on the Pull Request, I need a GitHub personal access token with permission to write the message to notify the contributor about the error. @Xkonti

### CODE REVIEW

**Comment by Xkonti** [comment]
@Pr0-C0der Looks good. You can use `PR_TOKEN` secret to get the token.

File: .github/workflows/pr-check.yml [hunk 1]
```diff
@@ -0,0 +1,47 @@
+# This workflow will build a golang project
+# For more information see: https://docs.github.com/en/actions/automating-builds-and-tests/building-and-testing-go
+
+name: Test PR
+
+on:
+  push:
+    branches:
+      - main
```
Comment (Xkonti) [inline .github/workflows/pr-check.yml:9]: This shouldn't be here. Merges to `main` are prohibited and ALL changes happen through PRs.

File: .github/workflows/pr-check.yml [hunk 2]
```diff
@@ -0,0 +1,47 @@
+# This workflow will build a golang project
```
Comment (Xkonti) [inline .github/workflows/pr-check.yml:1]: This workflow will build `and test` the library

**Review by Xkonti — CHANGES_REQUESTED** [review]
Nice job with the comment!

**Comment by Pr0-C0der** [comment]
Please review @Xkonti. I was having issues with commenting stuff, so I decided to remove it. Nevertheless, when 'go test' command runs it will automatically tell the user which test file failed and on which line.

**Pull request merged**